src/webgpu/api/operation/device/lost.spec.ts

```diff
@@ -2,24 +2,82 @@ export const description = `
 Tests for GPUDevice.lost.
 `;
 
-import { Fixture } from '../../../../common/framework/fixture.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { attemptGarbageCollection } from '../../../../common/util/collect_garbage.js';
+import { getGPU } from '../../../../common/util/navigator_gpu.js';
+import { assert, raceWithRejectOnTimeout } from '../../../../common/util/util.js';
+import { GPUTest } from '../../../gpu_test.js';
 
-export const g = makeTestGroup(Fixture);
+class DeviceLostTests extends GPUTest {
+  // Default timeout for waiting for device lost is 2 seconds.
+  readonly kDeviceLostTimeoutMS = 2000;
+
+  getDeviceLostWithTimeout(device: GPUDevice = this.device): Promise<GPUDeviceLostInfo> {
+    return raceWithRejectOnTimeout(
+      this.device.lost,
+      this.kDeviceLostTimeoutMS,
+      'device was not lost'
+    );
+  }
+
+  expectDeviceDestroyed(): void {
+    this.eventualAsyncExpectation(async niceStack => {
+      try {
+        const lost = await this.getDeviceLostWithTimeout();
+        this.expect(lost.reason === 'destroyed', 'device was lost from destroy');
+      } catch (ex) {
+        niceStack.message = 'device was not lost';
+        this.rec.expectationFailed(niceStack);
+      }
+    });
+  }
+}
+
+export const g = makeTestGroup(DeviceLostTests);
 
 g.test('not_lost_on_gc')
   .desc(
     `'lost' is never resolved by GPUDevice being garbage collected (with attemptGarbageCollection).`
   )
-  .unimplemented();
+  .fn(async t => {
+    // Create a new device for this because it is hard to lose all refs to the test fixture device.
+    const adapter = await getGPU().requestAdapter();
+    assert(adapter !== null);
+    let device: GPUDevice | undefined = await adapter.requestDevice();
+    assert(device !== null);
```

Comment:
It shouldn't be possible for device to be null (types affirm this because there's no `null` in `GPUDevice | undefined`)
```suggestion
```

Reply:
Done